core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1492,6 +1492,13 @@ abstract class RDD[T: ClassTag](
    * Save this RDD as a text file, using string representations of elements.
    */
   def saveAsTextFile(path: String): Unit = withScope {
+    saveAsTextFile(path, null)
+  }
+
+  /**
+   * Save this RDD as a compressed text file, using string representations of elements.
+   */
+  def saveAsTextFile(path: String, codec: Class[_ <: CompressionCodec]): Unit = withScope {
     // https://issues.apache.org/jira/browse/SPARK-2075
     //
     // NullWritable is a `Comparable` in Hadoop 1.+, so the compiler cannot find an implicit
```

Comment:
BTW, it seems we don't this workaround now. cc: @srowen

Reply:
Sure, this can be removed then -- I think we just need to remove this comment?

Reply:
oh, yes. This pr removes the workaround...

Reply:
@maropu @srowen Here you all think we should remove this comment, why? I don't understand, could you explain it more? Thank you so much!

Reply:
Have you checked SPARK-2075? We don't support hadoop 1.x anymore in master.

Reply:
I got it, thank you! @maropu